packages/gatsby/cache-dir/socketIo.js

```diff
@@ -52,10 +52,18 @@ export default function socketIo() {
               clearError(msg.payload.id)
             }
           }
+
           if (msg.type && msg.payload) {
             ___emitter.emit(msg.type, msg.payload)
           }
         })
+
+        // Prevents certain browsers spamming XHR 'ERR_CONNECTION_REFUSED'
+        // errors within the console, such as when exiting the develop process.
+        socket.on(`disconnect`, () => {
+          console.warn(`[socket.io] Disconnected from dev server.`)
+          socket.close()
+        })
```

Comment:
As I understand, after talking to @mxstbr, this check in socketIo.js is not really required since that's for a different connection than is being addressed in this PR. Please undo changes in this file.

Reply:
Still relevant, please note the comment explaining why.

The PR is fixing the main problem, by allowing the internal port to be configured. @mxstbr did also initially suggest handling the error issue by preventing the error spam when disconnected. That is also a problem without configuring the *server status port* for any developer that stops the server(or it crashes) and they want to view logs without seeing errors continue to fill it up.

If you want that as a separate PR instead, I guess I can do that?